### PR TITLE
Import cleaning (direct model imports; isort)

### PR DESCRIPTION
**Pull request by akx**

### CODE REVIEW

File: shoop/core/utils/reference.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from django.utils.encoding import force_text
 
 from shoop.utils.importing import load
-from shoop.core.models.counters import Counter, CounterType
+from shoop.core.models.counters import Counter, CounterType  # must be direct import
```
Comment (suutari-ai) [inline shoop/core/utils/reference.py]: Why this must be direct import?

File: shoop/front/apps/simple_search/views.py [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
 from django.utils.encoding import force_bytes
 from django.views.generic import ListView
-import hashlib
-from shoop.core.models.products import Product
-from django.core.cache import cache
-from shoop.front.utils.views import cache_product_things
+from shoop.core.models import Product
 from shoop.front.utils.product_sorting import PRODUCT_SORT_CHOICES, sort_products
+from shoop.front.utils.views import cache_product_things
+import hashlib
```
Comment (suutari-ai) [inline shoop/front/apps/simple_search/views.py]: While fixing import statement ordering make it conform PEP-8 groping too, i.e. first stdlib imports, then third party imports, then local imports (= shoop imports in our case), each group in alphabetical order and groups separated by a blank line.

**Comment by suutari-ai** [comment]
@akx: Hey, thanks. This is good stuff. :) Can you reply to the couple inline comments above and rebase this, so I may merge?

File: shoop/api/factories.py [hunk 3]
```diff
@@ -5,19 +17,10 @@
...
 from shoop.api._vendor.parler_rest.fields import TranslatedFieldsField
 from shoop.api._vendor.parler_rest.serializers import TranslatableModelSerializer
-from rest_framework import serializers, viewsets, permissions
 
-
-def serializer_factory(model, serializer_class=None, attrs=None, meta=None):
-    attrs = attrs or {}
-    meta = meta or {}
-    if not serializer_class:
-        if issubclass(model, TranslatableModel):
```
Comment (suutari-ai) [inline shoop/api/factories.py]: WTF happened here?

File: shoop/core/fields/tagged_json.py [hunk 4]
```diff
@@ -5,6 +5,20 @@
...
+from django.core.exceptions import ImproperlyConfigured
+from django.utils import six
+from jsonfield.encoder import JSONEncoder
+from six import text_type
+
+from shoop.utils.importing import load
+from shoop.utils.iterables import first
+
 """
 "Tagged JSON" encoder/decoder.
```
Comment (suutari-ai) [inline shoop/core/fields/tagged_json.py]: This is not correct. The module docstring should go above the imports. Quote from PEP-8:
> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

Comment (suutari-ai) [inline shoop/core/fields/tagged_json.py]: Maybe the reason why isort does this is because the unicode_literals import is too soon?

**Comment by akx** [comment]
There, I think that's about it :)

File: shoop/core/fields/tagged_json.py (hunk 4 above)
Comment (suutari-ai) [inline shoop/core/fields/tagged_json.py]: This is interesting... If the imports go before the module docstring, it won't be treated as module docstring at all. Also the type of module docstring in Python 2 is unicode even when the unicode_literals import follows the docstring.
```
$ cat a.py 
"""
Docstring.
"""

from __future__ import unicode_literals
import sys

print(sys.version, type(__doc__))
$ cat b.py 
from __future__ import unicode_literals
import sys

"""
Docstring.
"""

print(sys.version, type(__doc__))
$ python a.py
('2.7.6 (default, Jun 22 2015, 17:58:13) \n[GCC 4.8.2]', <type 'unicode'>)
$ python b.py
('2.7.6 (default, Jun 22 2015, 17:58:13) \n[GCC 4.8.2]', <type 'NoneType'>)
$ python3 a.py
3.4.0 (default, Jun 19 2015, 14:20:21) 
[GCC 4.8.2] <class 'str'>
$ python3 b.py
3.4.0 (default, Jun 19 2015, 14:20:21) 
[GCC 4.8.2] <class 'NoneType'>
```

**Comment by akx** [comment]
retest this please

**Comment by akx** [comment]
retest this please

**Comment by akx** [comment]
Well *that's* more like it.

**Comment by akx** [comment]
(Self-)merging this per Slack discussion now that tests pass.

**Pull request merged**